### PR TITLE
Implementing Trip Report with attribute formating

### PR DESCRIPTION
**Pull request by mail2bishnoi**
Hi Anton,

I have started creating the remaining reports. I started it with the Trips report. Along with this, I have also started customizing attributes such as distance, speed, and duration, etc. on the basis of user preference. Therefore, please review the code and suggest improvements in it.

Thanks

### CODE REVIEW

File: modern/src/common/formatter.js [hunk 1]
```diff
@@ -27,12 +29,44 @@ export const formatPosition = (value, key) => {
...
+        return `${(value * 1).toFixed(2)} ${t('sharedKn')}`;
+  }  
+};
+
+export const formatHours = (value) => {
+  let hours, minutes;
+  hours = Math.floor(value / 3600000);
+  minutes = Math.floor(value % 3600000 / 60000);
+
+  return `${hours} ${t('sharedHourAbbreviation')} ${minutes} ${t('sharedMinuteAbbreviation')}`;
```
Comment (tananaev) [inline modern/src/common/formatter.js:71]: Is there a way to do this formatting with `moment`?

File: modern/src/common/formatter.js [hunk 2]
```diff
@@ -9,6 +9,8 @@ export const formatPosition = (value, key) => {
     case 'fixTime':
     case 'deviceTime':
     case 'serverTime':
+    case 'startTime':
```
Comment (tananaev) [inline modern/src/common/formatter.js:12]: I'm actually thinking of breaking this down into separate formatters.

Also, start and end time is not part of position.

File: modern/src/reports/TripReportPage.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const ReportFilterForm = ({ onResult }) => {
+
+  const handleSubmit = async (deviceId, from, to) => {
+    const query = new URLSearchParams({
+      deviceId,
+      from: from.toISOString(),
+      to: to.toISOString(),
+    });
+    const response = await fetch(`/api/reports/trips?${query.toString()}`, { headers: { Accept: 'application/json' } });
+    if(response.ok) {
```
Comment (tananaev) [inline modern/src/reports/TripReportPage.js:18]:
```suggestion
    if (response.ok) {
```

File: modern/src/reports/TripReportPage.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  }
+  return <ReportFilter handleSubmit={handleSubmit} />;
+}
+
+const TripReportPage = () => {
+  const distanceUnit = useAttributePreference('distanceUnit');
+  const speedUnit = useAttributePreference('speedUnit');
+  const [items, setItems] = useState([]);
+  return (
```
Comment (tananaev) [inline modern/src/reports/TripReportPage.js:29]:
```suggestion

  return (
```

**Review by tananaev — COMMENTED** [review]
Added some comments. Please also remove empty files.

**Comment by mail2bishnoi** [comment]
Hi Anton,

I have included your suggestions. Along with this, I have also created all the remaining reports except charts. Please review the code and suggest improvements.

Thanks.

File: modern/src/common/formatter.js (hunk 2 above)
Comment (tananaev) [reply modern/src/common/formatter.js:12]: Don't forget to remove this.

File: modern/src/reports/DailySummaryReportPage.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    });
+    const response = await fetch(`/api/reports/summary?${query.toString()}`, { headers: { Accept: 'application/json' } });
+    if (response.ok) {
+      onResult(await response.json());
+    }
+  }
+  return <ReportFilter handleSubmit={handleSubmit} />;
+}
+
+const DailySummaryReportPage = () => {
```
Comment (tananaev) [inline modern/src/reports/DailySummaryReportPage.js:26]: I don't think we need a separate page for this. Maybe we can pass a parameter to the summary report?

**Review by tananaev — COMMENTED** [review]
Add a couple of minor comments, but overall looks good and almost ready to be merged. Thanks for working on this.

File: modern/src/reports/DailySummaryReportPage.js (hunk 5 above)
Comment (mail2bishnoi) [reply modern/src/reports/DailySummaryReportPage.js:26]: I thought that too. But created a separate page to sync with the report type Dropdown of the old web. Although now removed.

**Comment by mail2bishnoi** [comment]
Hi Anton,

I have included your suggestions and removed the DailySummary Report page and done minor code fixes.

**Comment by tananaev** [comment]
Merged, thank you so much for working on this.

**Pull request merged**